### PR TITLE
C API for renumbering the samples

### PR DESCRIPTION
**Pull request by ChuckHastings**
Definition of C API for renumbering of sampling results.

### CODE REVIEW

File: cpp/src/c_api/uniform_neighbor_sampling.cpp [hunk 1]
```diff
@@ -226,6 +229,22 @@ struct uniform_neighbor_sampling_functor : public cugraph::c_api::abstract_funct
                                                             vertex_partition_lasts,
                                                             do_expensive_check_);
 
+      std::optional<rmm::device_uvector<vertex_t>> renumber_map{std::nullopt};
+      std::optional<rmm::device_uvector<size_t>> renumber_map_offsets{std::nullopt};
+
+#if 0
```
Comment (alexbarghi-nv) [inline cpp/src/c_api/uniform_neighbor_sampling.cpp:235]: I assume this is disabled pending Seunghwa's PR getting merged?

**Review by alexbarghi-nv — CHANGES_REQUESTED** [review]

File: cpp/src/c_api/uniform_neighbor_sampling.cpp [hunk 2]
```diff
@@ -226,6 +229,22 @@ struct uniform_neighbor_sampling_functor : public cugraph::c_api::abstract_funct
...
+      std::tie(src, dst, renumber_map, renumber_map_offsets) =
+        cugraph::renumber_sampled_edgelist(handle_,
+                                           std::move(src),
+                                           hop ? std::make_optional(raft::device_span<int32_t const>{hop->data(), hop->size()}) : std::nullopt,
+                                           std::move(dst),
+                                           std::make_optional(std::make_tuple(raft::device_span<label_t const>{edge_label->data(),
+                                                                                                               edge_label->size()},
+                                               raft::device_span<size_t const>{offsets->data(),
+                                                                               offsets->size()}))
+                                           do_expensive_check);
```
Comment (alexbarghi-nv) [inline cpp/src/c_api/uniform_neighbor_sampling.cpp:245]:
```suggestion
                                           ,do_expensive_check_);
```

**Review by alexbarghi-nv — CHANGES_REQUESTED** [review]

File: cpp/src/c_api/uniform_neighbor_sampling.cpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/src/c_api/uniform_neighbor_sampling.cpp]: Yes.  Just pushed an update with a version of Seunghwa's branch merged in (not the latest).

File: cpp/src/c_api/uniform_neighbor_sampling.cpp (hunk 2 above)
Comment (ChuckHastings) [reply cpp/src/c_api/uniform_neighbor_sampling.cpp]: Fixed once I debugged :-)

**Review by alexbarghi-nv — APPROVED** [review]
Tested, working 👍

**Review by seunghwak — APPROVED** [review]
LGTM.

**Pull request merged**